### ISSUE TITLE
Error in enum Python class generation

### ISSUE THREAD

**Issue by gaurav** [linkml/linkml#254]
There's an error in enum Python class generation that I've found while trying to get PR #239 to work. An existing test (the "kitchen_sink" generator test) catches this.

Given the following schema:
```
enums:
  FamilialRelationshipType:
    permissible_values:
      SIBLING_OF:
      PARENT_OF:
      CHILD_OF:
```

I can generate the Python data class for this. However, I find that:

```
print(kitchen_module.FamilialRelationshipType('SIBLING_OF')) # Produces: 'SIBLING_OF: '
```

I'm not sure why the extraneous `: ` is being added, but this is what is causing PR #239 to fail: https://github.com/linkml/linkml/pull/239/checks?check_run_id=2968179446#step:6:410

**Comment by hsolbrig** [comment]
Ooops - will look at it in this context

**Comment by gaurav** [comment]
Thanks, Harold! Let me know if you'd like me to write up a PR that demonstrates this issue.

**Comment by hsolbrig** [comment]
The reason it is doing what is doing is that it is replicating the input ('SIBLING_OF:').   https://github.com/linkml/linkml-runtime/blob/main/linkml_runtime/utils/enumerations.py#L99 very explicitly tries to emit exactly what you are seeing.

Where is the `str` function being used that breaks things?   Should it be using kitchen_module.FamilialRelationshipType('SIBLING_OF').code instead?

**Comment by gaurav** [comment]
I think you might be right! Have a look at this code fragment:

https://github.com/linkml/linkml-runtime/blob/b9dd9b24f6c9cbfe9adb7a2c9b5bfb8c07293d03/linkml_runtime/utils/enumerations.py#L99-L104

Am I right in thinking that `getattr(self._code, f.name)` should actually be `getattr(self._code.text, f.name)`, as it is in the `__str__()` method?

If that's not it, I did write a small piece of code that triggers the error: https://github.com/linkml/linkml/commit/b911b14bc38240063e2d258df7a010444a97371d

And you can see the result in the GitHub action testing log here: https://github.com/linkml/linkml/pull/239/checks?check_run_id=2995155282#step:6:450

**Comment by hsolbrig** [comment]
@gaurav  - Can we close this issue?

**Comment by gaurav** [comment]
Yup, we implemented this in PR linkml/linkml-runtime#17. Sorry for not closing it sooner!
